include/seqan3/argument_parser/detail/format_base.hpp

```diff
@@ -0,0 +1,219 @@
+                                       "CHAR",
+                                       "STRING"};
+
+        if constexpr (meta::in<types, type>::value)
+            return names[meta::find_index<types, type>::value];
+        else
+            return "UNKOWN_TYPE";
+    }
+
+    /*!\brief Returns the value_type of the input container as a string (reflection).
+     * \tparam container_type The container type for which to query it's value_type.
+     */
+    template <typename container_type>
+    requires container_concept<container_type> && !std::is_same_v<container_type, std::string>
+    inline std::string get_type_name_as_string(container_type const & /**/)
+    {
+        typename container_type::value_type tmp;
+        return get_type_name_as_string(tmp);
+    }
+
+
+    /*!\brief Formats the type of a value for the help page printing.
+     *
+     * \param value The value to deduct the type from.
+     */
+    template <typename option_value_type>
+    inline std::string option_type_and_list_info(option_value_type const & value)
+    {
+        return ("\\fI" + get_type_name_as_string(value) + "\\fP");
+    }
+
+    /*!\brief Formats the container and its value_type for the help page printing.
+     *
+     * \param container The container to deduct the type from.
+     */
+    template <typename container_type>
+    requires container_concept<container_type> && !std::is_same_v<container_type, std::string>
+    inline std::string option_type_and_list_info(container_type const & container)
+    {
+        return ("List of \\fI" + get_type_name_as_string(container) + "\\fP's");
+    }
+
+    /*!\brief Formats the option/flag indentifier pair for the help page printing.
+     *
+     * \param short_id The short identifier of the option/flag.
+     * \param long_id  The long identifier of the option/flag.
+     *
+     * \details
+     * e.g. "-i,--integer","-i", or "--integer".
+     */
+    inline std::string prep_id_for_help(char const short_id, std::string const & long_id)
+    {
+        // Build list item term.
+        std::string term;
+        if (short_id != '\0')
+        {
+            term = "\\fB-" + std::string(1, short_id) + "\\fP";
+        }
+        if (short_id != '\0' && !long_id.empty())
+            term.append(", ");
+        if (!long_id.empty())
+        {
+            term.append("\\fB--" + long_id + "\\fP");
+        }
+        return term;
+    }
+
+    /*!\brief Excapes certain characters for correct output.
+     *
+     * \param original The string containing characters tobe escaped.
+     *
+     * \details
+     * Special characters considered are `"`, `\`, `&`, `<` and `>`.
+     */
+    std::string escape_special_xml_chars(std::string const & original)
+    {
+        std::string escaped;
+        escaped.resize(original.size()); // will be as least as long
+
+        for (auto c : original)
+        {
+            if (c == '"')
+                escaped.append("&quot;");
```

Comment:
Hm it was called like that already in seqan2. I don't know actually, maybe both?